addons/web/static/src/webclient/menus/menu_service.js

```diff
@@ -23,6 +25,13 @@ function makeFetchLoadMenus() {
 
 function makeMenus(env, menusData, fetchLoadMenus) {
     let currentAppId;
+    function evaluateInvisible() {
+        Object.values(menusData).forEach((menu) => {
+            menu.invisible = evaluateBooleanExpr(menu.webInvisible, user.evalContext);
+            delete menu.webInvisible;
+        });
+    }
+    evaluateInvisible();
```

Comment:
3.6ms with all addons, I'd say we can move forward with this, as we plan to rework the menu logic right after anyway